File: Core/DomainLayer/Models/ActivityFeed.cs
using System;

namespace IntelliFit.Domain.Models
{
    public class ActivityFeed
    {
        public int ActivityId { get; set; }
        public int UserId { get; set; }
        public string ActivityType { get; set; } = null!;
        public string Title { get; set; } = null!;
        public string? Description { get; set; }
        public string? Icon { get; set; }
        public int? ReferenceId { get; set; }
        public string? ReferenceType { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public virtual User User { get; set; } = null!;
    }
}

File: codeflex-ai/src/lib/api/activityFeed.ts
import { apiFetch, type ApiResponse } from './client';

export interface ActivityFeedDto {
  activityId: number;
  userId: number;
  userName: string;
  activityType: string;
  title: string;
  description: string;
  category: string;
  relatedEntityId?: number;
  createdAt: string;
}

export interface CreateActivityFeedDto {
  userId: number;
  activityType: string;
  title: string;
  description: string;
  category?: string;
  relatedEntityId?: number;
}

export const activityFeedApi = {
  /**
   * Create activity
   */
  async createActivity(data: CreateActivityFeedDto): Promise<ApiResponse<ActivityFeedDto>> {
    return apiFetch<ActivityFeedDto>('/activity-feed', {
      method: 'POST',
      body: JSON.stringify(data),
    });
  },

  /**
   * Get user's activities
   */
  async getUserActivities(userId: number, limit: number = 50): Promise<ApiResponse<ActivityFeedDto[]>> {
    return apiFetch<ActivityFeedDto[]>(`/activity-feed/user/${userId}?limit=${limit}`);
  },

  /**
   * Get recent activities (all users - for social feed)
   */
  async getRecentActivities(limit: number = 100): Promise<ApiResponse<ActivityFeedDto[]>> {
    return apiFetch<ActivityFeedDto[]>(`/activity-feed/recent?limit=${limit}`);
  },

  /**
   * Delete activity
   */
  async deleteActivity(id: number): Promise<ApiResponse<void>> {
    return apiFetch<void>(`/activity-feed/${id}`, {
      method: 'DELETE',
    });
  },
};
